client/finality-grandpa/src/voting_rule.rs

```diff
@@ -227,6 +227,11 @@ impl<Block, B> VotingRule<Block, B> for VotingRules<Block, B> where
 				if let Some(header) = rule
 					.restrict_vote(backend.clone(), &base, &best_target, &restricted_target)
 					.await
+					.filter(|(_, restricted_number)| {
+						// we can only restrict votes within the interval [base, target]
+						restricted_number >= base.number()
+							&& restricted_number < restricted_target.number()
```

Comment:
```suggestion
							&& restricted_number <= restricted_target.number()
```

Either this or the comment above is wrong :D

Reply:
I updated the comment to make it clear why we do this :)

Reply:
I meant `[base, target]` to `[base, target)` :D

Reply:
Yeah I know, my point was that returning `target` would be valid but just unnecessary for us to fetch again. But I updated the comment to say `[base, target)` since that's simpler anyway.